pySDC/core/Problem.py

```diff
@@ -25,18 +25,17 @@ class ptype(object):
         dtype_f: RHS data type
     """
 
-    def __init__(self, init, dtype_u, dtype_f, params):
+    def __init__(self, init, dtype_u, dtype_f, **kwargs):
         """
-        Initialization routine
+        Initialization routine.
+        Add the problem parameters as keyword arguments.
 
         Args:
             init: number of degrees-of-freedom (whatever this may represent)
             dtype_u: variable data type
             dtype_f: RHS data type
-            params (dict): set or parameters
         """
-
-        self.params = _Pars(params)
+        self.params = _Pars(kwargs)
```

Comment:
I like the frozen classes because they prevent a bug where you misspell some variable and the desired effect does not happen. Robert implemented these frozen classes because a bug like this was plaguing him and I have to admit that the same thing happened to me multiple times. By doing it here, we don't need to import the frozen class thing in the individual problem classes, so I prefer it here. Maybe we can write better documentation, though. Would you be fine with that?